godot-macros/src/docs/md2bb.rs

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+//! Converts [markdown](https://en.wikipedia.org/wiki/Markdown) to [BBCode](https://en.wikipedia.org/wiki/BBCode).
+use markdown::mdast::Node;
+use markdown::{to_mdast, ParseOptions};
+use std::collections::HashMap;
+
+pub fn convert(md: &str) -> String {
+    let n = to_mdast(md, &ParseOptions::gfm()).unwrap();
+    let definitions = n
+        .children()
+        .unwrap()
+        .iter()
+        .filter_map(|n| match n {
+            Node::Definition(definition) => {
+                Some((definition.identifier.clone(), definition.url.clone()))
+            }
+            _ => None,
+        })
+        .collect::<HashMap<String, String>>();
+
+    n.children()
+        .unwrap()
+        .iter()
+        .filter_map(|n| node(n, &definitions))
+        .reduce(|a, b| format!("{a}\n{b}"))
```

Comment:
i'd just do `.collect::<String>().join("\n")` instead of this reduce, it better communicates intent imo

Reply:
you mean `collect::<Vec<_>>().join("\n")`?

Reply:
you mean `collect::<Vec<_>>().join("\n")`?